presto-docs/src/main/sphinx/spill/overview.rst

```diff
@@ -0,0 +1,85 @@
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Overview
+--------
+
+In the case of memory intensive operations, Presto allows offloading
+intermediate operator results to disk. The goal of this mechanism is to
+enable execution of queries that require amounts of memory exceeding per query
+or per node limits.
+
+The mechanism is similar to OS level page swapping. However, it is
+implemented on the application level to address specific needs of Presto.
+
+Properties related to spilling are described in :ref:`tuning-spilling`.
+
+
+Memory management and spill
+---------------------------
+
+By default, Presto kills queries if the memory requested by the query execution
+exceeds session properties ``query_max_memory`` or
+``query_max_memory_per_node``. This mechanism ensures fairness in allocation
+memory to queries and prevents deadlock caused by memory allocation.
+It is efficient when there is a lot of small queries in the cluster, but
+leads to killing large queries that don't stay within the limits.
+
+To overcome this inefficiency, the concept of revocable memory was introduced. A
+query can request memory that does not count toward the limits, but this memory
+can be revoked by the memory manager at any time. When memory is revoked, the
+query runner spills intermediate data from memory to disk and continues to
+process it later.
+
+In practice, when the cluster is idle, a memory intensive query may use all of
+the memory in the cluster. On the other hand, when the cluster does not have
+much free memory, the same query may be forced to use disk as storage for
+intermediate data. A query that is forced to spill to disk may have a longer
+execution time by orders of magnitude than a query that runs completely in
+memory.
+
+Please note that enabling spill-to-disk does not guarantee execution of all
+memory intensive queries. It is still possible that the query runner will fail
+to divide intermediate data into chunks small enough that every chunk fits into
+memory, leading to `Out of memory` errors while loading the data from disk.
+
+Supported operations
+------------------------
+
+Not all operations support spilling to disk, and each handles spilling
+differently. Currently, the mechanism is implemented for the following
+operations.
+
+Joins
+^^^^^
+
+During join operation, memory is used to perform lookups for processed rows of
```

Comment:
You might want to add a link to the optimizer docs about joins (https://teradata.github.io/presto/docs/current/optimizer/join-distribution-type.html) The lookup part is a bit unclear, here's a suggestion on how to reword it:
During the join operation, one of the tables being joined is stored in memory. This table is called the build table. The rows from the other table stream through and are passed onto the next operation if they match rows in the build table. The most memory-intensive part of the join is this build table.